src/hotspot/cpu/ppc/c1_LIRAssembler_ppc.cpp

```diff
@@ -2839,25 +2839,27 @@ void LIR_Assembler::negate(LIR_Opr left, LIR_Opr dest, LIR_Opr tmp) {
 
 void LIR_Assembler::rt_call(LIR_Opr result, address dest,
                             const LIR_OprList* args, LIR_Opr tmp, CodeEmitInfo* info) {
-  // Stubs: Called via rt_call, but dest is a stub address (no function descriptor).
+  // Stubs: Called via rt_call, but dest is a stub address (no FunctionDescriptor).
   if (dest == Runtime1::entry_for(C1StubId::register_finalizer_id) ||
-      dest == Runtime1::entry_for(C1StubId::new_multi_array_id   )) {
+      dest == Runtime1::entry_for(C1StubId::new_multi_array_id   ) ||
+      dest == Runtime1::entry_for(C1StubId::is_instance_of_id    )) {
```

Comment:
Should there be an assertion that `dest` is in the CodeCache?
Or even use that check as condition to emit the optimized call?

Reply:
Yes, maybe that's better.